lib/srv/alpnproxy/conn_test.go

```diff
@@ -165,15 +164,15 @@ func TestPingConnection(t *testing.T) {
 				for {
 					n, err := r.Read(buf)
 					if err != nil {
-						switch {
+						switch err {
```

Comment:
Note that these changes are not strictly necessary.

`errors.Is(io.EOF, io.EOF)` will return true.

Reply:
It will, but it will also return true for `errors.Is(trace.Wrap(io.EOF), io.EOF)`. This switch (at least from my testing) will pass through `trace.Wrap`ed errors, which will then fail the test later on.